src/Services/SpreadSheetService.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * @package OpenEMR
+ * @link    http://www.open-emr.org
+ * @author  Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @author  Stephen Waite <stephen.waite@cmsvt.com>
+ * @copyright Copyright (c) 2021-2022  Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @copyright Copyright (c) 2022 Stephen Waite <stephen.waite@cmsvt.com>
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ *
+ * This middleware is to allow a uniform use of the spreadsheet library.
+ * By passing an array values, a spreadsheet can be dropped to the client from anywhere
+ * To use this service send the file name and an array like this
+ * The first row is the headers that can be dynamcally generated or static
+ * $sheet_array = [
+ *     [NULL, 2010, 2011, 2012],
+ *     ['Q1',   12,   15,   21],
+ *     ['Q2',   56,   73,   86],
+ *     ['Q3',   52,   61,   69],
+ *     ['Q4',   30,   32, 'Harry'],
+ *     ];
+ */
+
+namespace OpenEMR\Services;
+
+use OpenEMR\Common\Logging\SystemLogger;
+use PhpOffice\PhpSpreadsheet\{
+    IOFactory,
+    Helper\Sample,
+    Spreadsheet,
+};
+
+class SpreadSheetService extends Spreadsheet
+{
+    private array $arrayData;
+    private string $fileName;
+    private array $header;
+    private array $row;
+    private array $fields;
+
+    public function __construct(
+        $arrayData,
+        $fileName = 'report',
+        $fields
```

Comment:
i think you may get a php warning here since parameters with default setting are supposed to go after parameters without default settings (although maybe __construct is special and won't throw the warning)

Reply:
didn't see a warning but will switch since makes sense